public/app/plugins/datasource/loki/metric_find_query.ts

```diff
@@ -0,0 +1,35 @@
+import { LokiDatasource, LOKI_ENDPOINT, LEGACY_LOKI_ENDPOINT } from './datasource';
+
+export async function processMetricFindQuery(datasource: LokiDatasource, query: string) {
```

Comment:
Just out of curiosity why is this not in the datasource? Why does it need to be in a separate file?

Reply:
Good question. I was following Prometheus file structure, where it was implemented in its own file. However, in Prometheus we are creating class, which is not a case with Loki. So after thinking more about this, and after looking at more examples how different datasources implement `metricFindQuery`, I have decided to move it directly to datasource, as is doesn't have valid reason to be outside the datasource. I have also moved tests to `datasource.test.ts`.